### PR TITLE
HOSTEDCP-1020: Add Create Cluster for KubeVirt for HCP CLI

### PR DESCRIPTION
**Pull request by bryan-cox**
**What this PR does / why we need it**:
Add the create cluster command for the KubeVirt platform for the HCP CLI.

**Which issue(s) this PR fixes**:
Fixes [HOSTEDCP-1020](https://issues.redhat.com/browse/HOSTEDCP-1020).

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes unit tests.

### CODE REVIEW

**Comment by bryan-cox** [comment]
@davidvossel, @nirarg - are there any flags you would want removed for the productized HyperShift CLI (`hcp create cluster kubevirt`)?

Also, do you want any changes to `applyPlatformSpecificsValues`? If not, I can export the original function and just reference it in the productized CLI.

File: product-cli/cmd/cluster/kubevirt/create.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+		APIServerAddress:          "",
+		Memory:                    "8Gi",
+		Cores:                     2,
+		ContainerDiskImage:        "",
+		RootVolumeSize:            32,
+		InfraKubeConfigFile:       "",
+		CacheStrategyType:         "",
+	}
+
+	cmd.Flags().StringVar(&opts.KubevirtPlatform.APIServerAddress, "api-server-address", opts.KubevirtPlatform.APIServerAddress, "The API server address that should be used for components outside the control plane")
```
Comment (davidvossel) [inline product-cli/cmd/cluster/kubevirt/create.go:41]: this `api-server-address` flag can be removed

File: product-cli/cmd/cluster/kubevirt/create.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+	cmd.Flags().Uint32Var(&opts.KubevirtPlatform.Cores, "cores", opts.KubevirtPlatform.Cores, "The number of cores inside the vmi, Must be a value greater than or equal to 1")
+	cmd.Flags().StringVar(&opts.KubevirtPlatform.RootVolumeStorageClass, "root-volume-storage-class", opts.KubevirtPlatform.RootVolumeStorageClass, "The storage class to use for machines in the NodePool")
+	cmd.Flags().Uint32Var(&opts.KubevirtPlatform.RootVolumeSize, "root-volume-size", opts.KubevirtPlatform.RootVolumeSize, "The size of the root volume for machines in the NodePool in Gi")
+	cmd.Flags().StringVar(&opts.KubevirtPlatform.RootVolumeAccessModes, "root-volume-access-modes", opts.KubevirtPlatform.RootVolumeAccessModes, "The access modes of the root volume to use for machines in the NodePool (comma-delimited list)")
+	cmd.Flags().StringVar(&opts.KubevirtPlatform.ContainerDiskImage, "containerdisk", opts.KubevirtPlatform.ContainerDiskImage, "A reference to docker image with the embedded disk to be used to create the machines")
```
Comment (davidvossel) [inline product-cli/cmd/cluster/kubevirt/create.go:47]: this `containerdisk` flag can be removed

**Comment by davidvossel** [comment]
> @davidvossel, @nirarg - are there any flags you would want removed for the productized HyperShift CLI (hcp create cluster kubevirt)?

yes, I left a few comments there about cli args that can be removed.

> Also, do you want any changes to applyPlatformSpecificsValues? If not, I can export the original function and just reference it in the productized CLI.

We're good with the exact same defaults. thanks!

**Review by davidvossel — COMMENTED** [review]
/approve
/lgtm

**Comment by bryan-cox** [comment]
TestUpgradeControlPlane/EnsureNoCrashingPods flake.
/retest-required

**Review by davidvossel — COMMENTED** [review]
/lgtm

**Pull request merged**